### DISCUSSION TITLE
ModuleNotFoundError: No module named 'six'

### DISCUSSION THREAD

**Discussion by wpawlikowski-form3** [tbabej/taskwiki#330] [Community support]
Hi, just recently upgraded my two workstations (MacOS, homebrew) and got this on both. They both use vim and python3 from homebrew repo.

**Comment by pwsiegel** [comment]
Same thing happened to me.

**Comment by pwsiegel** [comment]
Alright, I was able to fix this on my machine.  Prior to running `brew upgrade` I had installed `tasklib` and the `six` dependency using `pip3 install tasklib`.  After updating everything I eventually deduced that vim was no longer looking in the virtual environment to which `pip3` installs things, and instead I had to run `pip3.8 install tasklib` and it started working again.  Tasklib wasn't the only thing to break - I also had to install `pipx` with `pip3.8` - so I think the underlying fault lies with the update of `python3`.

That being said, there might be something that taskwiki could do to help us with this issue.  I generally don't like having python dependencies installed globally on my system; I use virtual environments whenever possible.  In fact it is a testament to how much I love taskwiki that I ultimately consented to globally install tasklib - there are other tools that I decided not to adopt just to keep my global python distribution clean.  Nevertheless it would be very handy if taskwiki could ship with its own copy of tasklib, or at least to allow us to point to a specific virtualenv containing tasklib.  Is that possible?

**Comment by lingow** [comment]
I was experiencing the same issue. After installing https://github.com/jmcantrell/vim-virtualenv, vim could grab the virtualenv that was active in the environment, and taskwiki worked without any issue.

**Comment by webdog** [comment]
homebrew is using python3.9, which is causing this plugin to fail with the same errors as above, when using macvim

I resolved it by finding the path of the py3 interpreter in macvim:

`:py3 import sys`
`:print(sys.path)`

...and running that version of python's pip to resolve the missing dependency. Seems to to be working now.

**Comment by aloysb** [comment]
I have the same issue here, only with Macvim. Vim does work fine...
I've tried @pwsiegel solution but unfortunately it does not work.

Also tried the other solutions but it didn't fix the issue.

I've also reinstalled Macvim, no success.

Any clues ? I have not idea how to fix this.
I open my vimwiki in Vim to be able to use Taskwiki, but I'd prefer to work straight from Mvim.

**Comment by tbabej** [comment]
@Aloysb The solution basically should consist of two steps:
1. Figure out path to the Python binary that Macvim uses
2. Run `<path-to-that-python-binary> -m pip install -r <path-to-taskwiki-requirements.txt>`

**Comment by aloysb** [comment]
Alright, thank you so much.

The step you've indicated informed me that six was already install.
I tried to uninstall and reinstall, same issue.

I finally fixed the issue by running `brew upgrade python3` and then `brew
link --overwrite python`
I think I had to faff around python while fixing an issue with UltiSnips,
which somehow broke this.
(Don't ask me more, I'm waaaaaaaaaaaaaaaaaaay out of my depth here).

Bottom line is, it's fixed, it was somehow linked to Python and upgrading
it + following the prompt fixed the issue.

Thank you very much for the help.

On Tue, 5 Jan 2021 at 03:14, Tomas Babej <notifications@github.com> wrote:

> @Aloysb <https://github.com/Aloysb> The solution basically should consist
> of two steps:
>
>    1. Figure out path to the Python binary that Macvim uses
>    2. Run <path-to-that-python-binary> -m pip install -r
>    <path-to-taskwiki-requirements.txt>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/tools-life/taskwiki/issues/268#issuecomment-754340125>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEHYZ4VIY7NKRC2XZNPISJDSYJYWVANCNFSM4ND3TXWQ>
> .
>


-- 
BERGER Aloys
*aloysberger@gmail.com <aloysberger@gmail.com>*
*Mobile : 0499 249 253*

**Comment by VeniVidiVinzi** [comment]
Hi,
I had the same problem.
New version of brew-vim uses python 3.11 so I just had to reinstall following packages by `pip3.11 install XXX`:
- six
- packaging
- tasklib

Now everything works fine again.
Maybe this could help someone.
